lib/PuppeteerSharp/Browser.cs

```diff
@@ -70,14 +70,24 @@ public string WebSocketEndpoint
             return await target.Page();
         }
 
+        /// <summary>
+        /// Returns An Array of all active targets
+        /// </summary>
+        /// <returns>An Array of all active targets</returns>
+        public IEnumerable<Target> Targets() => _targets.Values.Where(target => target.IsInitialized);
+
+        /// <summary>
+        /// Returns a Task which resolves to an array of all open pages.
+        /// </summary>
+        /// <returns>Task which resolves to an array of all open pages.</returns>
         public async Task<IEnumerable<Page>> Pages()
```

Comment:
would it make sense to return an array instead? do we want this to be lazy?

Reply:
Absolutely, let's do it

Reply:
`PagesAsync`